### PR TITLE
Create Windows Test

### LINKED ISSUE #738: Create Windows Test in Matrix
There was a pickling error on windows in development for the file-watch and rebuild.

We need to run tests on Windows as well to make sure there are differences.

### PR DESCRIPTION
**Pull request by OsamaRab3**
Add a Windows testing  matrix. 
closes #738 

- [ ] :bug: (bug)
- [ ] :book: (Documentation)
- [ ] :arrow_up: (Update)
- [x] :dizzy: (New Feature)
- [ ] :radioactive: (Deprecation)
- [ ] :no_entry_sign: (Removal)
- [ ] :hammer_and_wrench: (Refactor)

## Changes have been Documented (If Applicable)

- [ ] YES - Changes have been documented

## Changes have been Tested

- [ ] YES - Changes have been tested

### CODE REVIEW

**Review by kjaymiller — CHANGES_REQUESTED** [review]
This is accurate but since the windows version consume resources that we would have to pay for if used too often I would request that you move that `runs-on` line to the `test-against-latest` section.

**Review by john0isaac — CHANGES_REQUESTED** [review]
What you did here is partially correct, you will need to create a matrix for the `test-against-latest` too that test against the latest windows and linux.
Right now your change removed the `linux-latest` tests completely.

See this resource about [matrix for GitHub actions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -34,8 +34,13 @@ jobs:
           python -m pytest
 
   test-against-latest:
-    # Always run against the latest version
-    runs-on: ubuntu-latest
+    # Always run against the latest version on both Windows and Linux
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ["3.12"]
```
Comment (john0isaac) [inline .github/workflows/test.yml:41]:
```suggestion
```

Comment (john0isaac) [reply .github/workflows/test.yml:41]: This is not being used here, removed.

**Review by john0isaac — APPROVED** [review]
Thank you for contributing LGTM!
@kjaymiller Please see if there anything that needs to be changed.

**Review by kjaymiller — APPROVED** [review]

**Comment by kjaymiller** [comment]
@allcontributors please add @OsamaRab3 for tests

**Pull request merged**